api/renterd/requests.go

```diff
@@ -44,3 +47,49 @@ type SyncerPeerResponse struct {
 type SyncerConnectRequest struct {
 	NetAddress string `json:"netAddress"`
 }
+
+// A ContractScanRequest contains the information of the host we are dialing
+// for contract related requests.
+type ContractScanRequest struct {
+	NetAddress string          `json:"netAddress"`
+	HostKey    types.PublicKey `json:"hostKey"`
+}
+
+// A ContractFormRequest requests that the host create a contract.
+type ContractFormRequest struct {
+	RenterKey   types.PrivateKey `json:"renterKey"`
+	NetAddress  string           `json:"netAddress"`
+	HostKey     types.PublicKey  `json:"hostKey"`
+	HostFunds   types.Currency   `json:"hostFunds"`
+	RenterFunds types.Currency   `json:"renterFunds"`
+	EndHeight   uint64           `json:"endHeight"`
+}
+
+// A ContractFormResponse is the response to /contract/form and
+// /contract/renew.  It contains the formed contract and the parent
+// transactions.
+type ContractFormResponse struct {
+	Contract rhp.Contract        `json:"contract"`
+	Parents  []types.Transaction `json:"parents"`
+}
```

Comment:
My gut says that renew should have a separate `Response` type, even if they currently contain the same fields.